plugins/lightspeed-backend/src/service/router.ts

```diff
@@ -0,0 +1,74 @@
+import { errorHandler } from '@backstage/backend-common';
+import { LoggerService } from '@backstage/backend-plugin-api';
+import { Config } from '@backstage/config';
+
+import express from 'express';
+import Router from 'express-promise-router';
+import { APIError, OpenAI } from 'openai';
+
+import { CompletionsRequestBody } from './types';
+import { validateCompletionsRequest } from './validation';
+
+export interface RouterOptions {
+  logger: LoggerService;
+  config: Config;
+}
+
+export async function createRouter(
+  options: RouterOptions,
+): Promise<express.Router> {
+  const { logger, config } = options;
+
+  const router = Router();
+  router.use(express.json());
+
+  router.get('/health', (_, response) => {
+    logger.info('PONG!');
+    response.json({ status: 'ok' });
+  });
+
+  router.post(
+    '/chat/completions',
+    validateCompletionsRequest,
+    async (request, response) => {
+      const { model, messages }: CompletionsRequestBody = request.body;
+      const baseURL = config.getString('lightspeed.baseURL');
+      const apiKey = config.getString('lightspeed.apiKey');
+
+      const openai = new OpenAI({
+        baseURL: baseURL,
+        apiKey: apiKey,
+      });
+
+      try {
+        const stream = await openai.chat.completions.create({
+          messages: messages,
+          model: model,
+          stream: true,
+        });
+
+        for await (const chunk of stream) {
```

Comment:
We would need the streaming support in the API as well so that we can show streaming data in the UI.

Reply:
Yes, I tried this approach in frontend plugin to read streamed data from backend (request has headers set to `'Content-Type': 'text/event-stream',`):
```
const reader = response.body!.pipeThrough(new TextDecoderStream())
      .getReader();

while (true) {
      const { value, done } = await reader.read();
      if (done) break;
      console.log('Received: ', value);`
}
```
Which logs data as they are streamed from backend.